Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -323,3 +355,32 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	}
 	return utilerrors.NewAggregate(allErrs)
 }
+
+// mustPrintWithKinds determines if printer output
+// should print multiple kinds of objects, even if
+// "all-kinds" flag is not set
+func mustPrintWithKinds(objs []runtime.Object, infos []*resource.Info, sorter *kubectl.RuntimeSort) bool {
+	var lastMap *meta.RESTMapping
+	var withKind bool
+
+	if withKind {
+		return withKind
+	}
+
+	for ix := range objs {
+		var mapping *meta.RESTMapping
+		if sorter != nil {
+			mapping = infos[sorter.OriginalPosition(ix)].Mapping
+		} else {
+			mapping = infos[ix].Mapping
+		}
+
+		// display "kind" column only if we have mixed resources
+		if lastMap != nil && mapping.Resource != lastMap.Resource {
+			withKind = true
```

Comment:
Once more we can apply the idea of terminating the loop as soon as we know our answer :)

If we see two resources of different kinds, we can return immediately:

```go
		if lastMap != nil && mapping.Resource != lastMap.Resource {
			return true
		}
		lastMap = mapping
	}
	return false
}
```

Notice we don't need the `withKind` variable at all.